tools/shared/src/main/scala/org/scalajs/core/tools/sourcemap/SourceMapWriter.scala

```diff
@@ -63,13 +63,20 @@ class SourceMapWriter(
 
   private var pendingColumnInGenerated: Int = -1
   private var pendingPos: Position = NoPosition
-  private var pendingName: Option[String] = None
+  private var pendingName: String = null
 
   writeHeader()
 
-  private def sourceToIndex(source: SourceFile) =
-    _srcToIndex.getOrElseUpdate(source,
-        (sources += sourceToURI(source)).size-1)
+  private def sourceToIndex(source: SourceFile): Int = {
+    if (_srcToIndex.contains(source)) {
+      return _srcToIndex(source)
```

Comment:
It is unnecessary but the bytecode generated is 2 bytes less. In this case this is not helping, so I'll remove it. It could potentially help with JIT inlining.